crates/oxc_ast_macros/src/lib.rs

```diff
@@ -1,30 +1,54 @@
 use proc_macro::TokenStream;
-use std::str::FromStr;
+use proc_macro2::TokenStream as TokenStream2;
+use quote::quote;
 
-/// Attach to AST node type (struct or enum), to signal to codegen to create visitor for this type.
-///
-/// Macro's role is not to generate code - it's purely a means to communicate information to the codegen.
-///
-/// Only thing macro does is add `#[derive(Ast)]` to the item.
-/// Deriving `Ast` does nothing, but supports `#[scope]`, `#[visit]`, and other attrs on struct fields.
-/// These "helper" attributes are also signals to the codegen, and do nothing in themselves.
+/// returns `#[repr(C, u8)]` if `enum_` has any non-unit variant,
+/// Otherwise it would return `#[repr(C)]`.
+fn enum_repr(enum_: &syn::ItemEnum) -> TokenStream2 {
+    if enum_.variants.iter().any(|var| !matches!(var.fields, syn::Fields::Unit)) {
+        quote!(#[repr(C, u8)])
+    } else {
+        quote!(#[repr(C)])

```

Comment:
`#[repr(u8)]` is what we want for fieldless enums.

```suggestion
        quote!(#[repr(u8)])
```